lib/Room.php

```diff
@@ -257,6 +262,11 @@ public function getType(): int {
 	}
 
 	public function setType(int $type): void {
+		// Force group room if public room was requested and public rooms are disallowed.
+		if (($type === self::TYPE_PUBLIC) && ($this->config->getAppValue('spreed', 'public_rooms_allowed', 'yes') === 'no')) {
+			$type = self::TYPE_GROUP;
+		}
```

Comment:
This is not correct. You want to do this change on the RoomService, not the model.
But see comment above.

Reply:
Modified places where choosen to avoid problems with lack of return errors handling. If you still recommend moving it elswhere - please suggest detailed change.

Reply:
As per above, you should do it only in the RoomService